pkg/controllers/capr/dynamicschema/dynamicschema.go

```diff
@@ -101,10 +118,12 @@ func addMachineTemplateSchema(name string, specSchema *schemas.Schema, allSchema
 		ID: templateSpecSchemaID,
 		ResourceFields: map[string]schemas.Field{
 			"template": {
-				Type: templateTemplateSpecSchemaID,
+				Type:        templateTemplateSpecSchemaID,
+				Description: "Template for creating new machines.",
 			},
 			"clusterName": {
-				Type: "string",
+				Type:        "string",
+				Description: "Name of the cluster (provisioning.cattle.io) that generated this template.",
```

Comment:
Nit: It may read better as `Name of the provisioning.cattle.io cluster object that generated this template.`